velox/exec/MergeSource.cpp

```diff
@@ -17,25 +17,79 @@
 #include "velox/exec/MergeSource.h"
 
 #include <boost/circular_buffer.hpp>
+#include "velox/common/testutil/TestValue.h"
 #include "velox/exec/Merge.h"
 #include "velox/vector/VectorStream.h"
 
+using facebook::velox::common::testutil::TestValue;
+
 namespace facebook::velox::exec {
 namespace {
 
+namespace {
+class PromiseNotifier {
+ public:
+  explicit PromiseNotifier(size_t initSize) {
+    promises_.reserve(initSize);
+  }
+
+  ~PromiseNotifier() {
+    for (auto& promise : promises_) {
+      promise.setValue();
+    }
+  }
+
+  void add(std::vector<ContinuePromise>&& promises) {
+    promises_.reserve(promises_.size() + promises.size());
+    for (auto& promise : promises) {
+      promises_.emplace_back(std::move(promise));
+    }
+    promises.clear();
+  }
+
+  void add(ContinuePromise&& promise) {
+    promises_.emplace_back(std::move(promise));
+  }
+
+ private:
+  std::vector<ContinuePromise> promises_;
+};
+
+void deferNotify(
```

Comment:
Why is this called deferNotify? It just delegates the notification of deferPromise to promiseNotifier

Reply:
It defers the notification.